jax/_src/numpy/lax_numpy.py

```diff
@@ -1665,26 +1666,29 @@ def _concatenate_array(arr, axis: int):
   return lax.reshape(arr, shape, dimensions)
 
 @_wraps(np.concatenate)
-def concatenate(arrays, axis: int = 0):
+def concatenate(arrays, axis: int = 0, dtype=None):
   if isinstance(arrays, (np.ndarray, ndarray)):
-    return _concatenate_array(arrays, axis)
+    return _concatenate_array(arrays, axis, dtype=dtype)
   _stackable(*arrays) or _check_arraylike("concatenate", *arrays)
   if not len(arrays):
     raise ValueError("Need at least one array to concatenate.")
   if ndim(arrays[0]) == 0:
     raise ValueError("Zero-dimensional arrays cannot be concatenated.")
   if axis is None:
-    return concatenate([ravel(a) for a in arrays], axis=0)
-  if hasattr(arrays[0], "concatenate"):
-    return arrays[0].concatenate(arrays[1:], axis)
+    return concatenate([ravel(a) for a in arrays], axis=0, dtype=dtype)
+  if hasattr(arrays[0], "concatenate") and (Version(np.__version__) >= Version(1.20)):
+    return arrays[0].concatenate(arrays[1:], axis, dtype=dtype)
```

Comment:
I don't think we should be branching based on numpy version within the implementation (the behavior of JAX should not change depending on what numpy version you happen to have installed!) it's appropriate to branch based on numpy version within the tests, however; you can see this in several existing tests: https://github.com/google/jax/blob/be8939771c9abbe384df1da5a970b306fad2e965/tests/lax_numpy_test.py#L57

Reply:
For what it's worth, the reason for the use of `concatenate` as a method here is to support `jnp.concatenate` called on PRNGKey arrays (see https://github.com/google/jax/pull/8381)

Reply:
We might need to modify this as part of this change: https://github.com/google/jax/blob/be8939771c9abbe384df1da5a970b306fad2e965/jax/_src/prng.py#L206-L209

Reply:
Ok so I'm looking into `jax.random.PRNGKeyArray`. I'm seeing that the `dtype` attribute is deprecated -- but also that it is always `uint32` even when x64 is enabled (or is this implementation dependent?). 
The most recent changes just raise an error in the `hasattr(arrays[0], "concatenate")` branch if the dtype isn't `uint32` or None.

Alternatively I can update the `jax.random.PRNGKeyArray.concatenate` method to accept the dtype -- and maybe sanitize there?

Reply:
> 

I don't see `numpy_version` being used anywhere in lax_numpy_test.py -- can we replace this with `Version(np.__version__)`?